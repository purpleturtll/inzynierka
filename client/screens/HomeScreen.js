import React from 'react';
import {
  View, Text, StyleSheet
} from 'react-native';

const HomeScreen = ({ navigation }) => {
  return(
    <View style={styles.container}>
    <View style={styles.header}>
    </View>
        <Text>Home</Text>
    </View>
  )
}
export default HomeScreen;

//#362893

const styles = StyleSheet.create({
  container: {
    flex:1,
    backgroundColor: '#fff'
  },

  button:{
    backgroundColor: '#E2E1E1',
    padding: 10,
    alignItems: 'center'
  }
});
