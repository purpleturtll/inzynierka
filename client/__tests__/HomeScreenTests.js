import 'react-native';
import React from 'react';
import {create, act} from 'react-test-renderer';
import HomeScreen, {onSeeMorePress} from '../screens/HomeScreen'

const screenTree = create(<HomeScreen />)

describe('HomeScreen component', () => {

    // Porównanie z ostatnim zrzutem z __tests__/__snapshots__
    it('snapshot', () => {
        expect(screenTree).toMatchSnapshot();
    });

    // Spradzenie poprawności zmiany stanu
    it.todo('handleFavChange');

    // Sprawdzenie czy lista zwierząt jest zdefiniowana - do poprawy (0 calli)
    it.todo('onSeeMorePress');
    /*
    it('onSeeMorePress', () => {
        const button = screenTree.root.findByProps({ testID: "SeeMoreButton"}).props;
        act(() => button.onPress());
        const mockedFn = jest.fn(() => onSeeMorePress());
        expect(mockedFn).toBeCalledWith(undefined, expect.anything()); 
    }); 
    */   
});