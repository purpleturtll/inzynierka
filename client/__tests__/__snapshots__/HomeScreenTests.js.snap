// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen component snapshot 1`] = `
<View>
  <RCTScrollView
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
  >
    <View>
      <View
        style={
          Object {
            "marginHorizontal": "5%",
            "marginVertical": "2%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderColor": "#555",
              "borderRadius": 20,
              "borderWidth": 1.7,
              "flexDirection": "row",
              "marginTop": 20,
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View
              style={
                Object {
                  "fontWeight": "bold",
                  "paddingBottom": 3,
                  "paddingLeft": 8,
                  "paddingTop": 6,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
          <TextInput
            allowFontScaling={true}
            collapsable={false}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            placeholder="Lorem ipsum ..."
            rejectResponderTermination={true}
            style={
              Object {
                "flex": 1,
                "paddingHorizontal": 8,
                "paddingVertical": 8,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View>
          <Text
            style={
              Object {
                "fontSize": 25,
                "paddingVertical": "8%",
              }
            }
          >
            Kogo szukasz?
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <RNGestureHandlerButton
              collapsable={false}
              onGestureEvent={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onHandlerStateChange={[Function]}
              rippleColor={0}
            >
              <View
                accessible={true}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "marginTop": "8%",
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#362893",
                      "borderRadius": 15,
                      "height": 80,
                      "marginBottom": 20,
                      "marginHorizontal": 10,
                      "marginTop": 45,
                      "width": 80,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingVertical": 28,
                        "textAlign": "center",
                        "textAlignVertical": "center",
                      }
                    }
                  >
                    Psy
                  </Text>
                </View>
                <Image
                  source={Object {}}
                  style={
                    Object {
                      "height": 60,
                      "position": "absolute",
                      "resizeMode": "cover",
                      "top": 0,
                      "width": 60,
                    }
                  }
                />
              </View>
            </RNGestureHandlerButton>
            <RNGestureHandlerButton
              collapsable={false}
              onGestureEvent={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onHandlerStateChange={[Function]}
              rippleColor={0}
            >
              <View
                accessible={true}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "marginTop": "8%",
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#362893",
                      "borderRadius": 15,
                      "height": 80,
                      "marginBottom": 20,
                      "marginHorizontal": 10,
                      "marginTop": 45,
                      "width": 80,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingVertical": 28,
                        "textAlign": "center",
                        "textAlignVertical": "center",
                      }
                    }
                  >
                    Koty
                  </Text>
                </View>
                <Image
                  source={Object {}}
                  style={
                    Object {
                      "height": 60,
                      "position": "absolute",
                      "resizeMode": "cover",
                      "top": 0,
                      "width": 60,
                    }
                  }
                />
              </View>
            </RNGestureHandlerButton>
            <RNGestureHandlerButton
              collapsable={false}
              onGestureEvent={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onHandlerStateChange={[Function]}
              rippleColor={0}
            >
              <View
                accessible={true}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "marginTop": "8%",
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#362893",
                      "borderRadius": 15,
                      "height": 80,
                      "marginBottom": 20,
                      "marginHorizontal": 10,
                      "marginTop": 45,
                      "width": 80,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingVertical": 28,
                        "textAlign": "center",
                        "textAlignVertical": "center",
                      }
                    }
                  >
                    Inne
                  </Text>
                </View>
                <Image
                  source={Object {}}
                  style={
                    Object {
                      "height": 60,
                      "position": "absolute",
                      "resizeMode": "cover",
                      "top": 0,
                      "width": 60,
                    }
                  }
                />
              </View>
            </RNGestureHandlerButton>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "paddingVertical": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "justifyContent": "flex-start",
                  }
                }
              >
                Ostatnio dodane
              </Text>
            </View>
            <RNGestureHandlerButton
              collapsable={false}
              onGestureEvent={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onHandlerStateChange={[Function]}
              rippleColor={0}
              testID="SeeMoreButton"
            >
              <View
                accessible={true}
                style={
                  Object {
                    "flex": 1,
                    "opacity": 1,
                    "paddingVertical": 5,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#777",
                      "fontSize": 15,
                      "justifyContent": "flex-end",
                      "paddingTop": 3,
                    }
                  }
                >
                  Zobacz więcej
                </Text>
              </View>
            </RNGestureHandlerButton>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 15,
                  "elevation": 6,
                  "marginVertical": 20,
                  "overflow": "hidden",
                  "shadowColor": "#fff",
                  "shadowOffset": Object {
                    "height": 10,
                    "width": 10,
                  },
                  "shadowOpacity": 0.5,
                  "shadowRadius": 5,
                }
              }
            >
              <RNGestureHandlerButton
                collapsable={false}
                onGestureEvent={[Function]}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                onHandlerStateChange={[Function]}
                rippleColor={0}
              >
                <View
                  accessible={true}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Image
                    source={Object {}}
                    style={
                      Object {
                        "height": 200,
                        "resizeMode": "cover",
                        "width": 300,
                      }
                    }
                  />
                </View>
              </RNGestureHandlerButton>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "padding": 10,
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "paddingBottom": 5,
                        "paddingTop": 3,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "fontSize": 16,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      Angus
                       
                    </Text>
                    <Text
                      style={
                        Object {
                          "fontSize": 16,
                        }
                      }
                    >
                      6/3/21
                    </Text>
                  </View>
                  <Text>
                    mieszaniec
                  </Text>
                  <Text>
                    samiec
                    , 
                    13
                     lat, 
                    9
                     kg
                  </Text>
                </View>
                <RNGestureHandlerButton
                  collapsable={false}
                  onGestureEvent={[Function]}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  onHandlerStateChange={[Function]}
                  rippleColor={0}
                >
                  <View
                    accessible={true}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "marginLeft": "35%",
                          "marginTop": 18,
                        }
                      }
                    >
                      <Text />
                    </View>
                  </View>
                </RNGestureHandlerButton>
              </View>
            </View>
          </View>
          <View>
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 15,
                  "elevation": 6,
                  "marginVertical": 20,
                  "overflow": "hidden",
                  "shadowColor": "#fff",
                  "shadowOffset": Object {
                    "height": 10,
                    "width": 10,
                  },
                  "shadowOpacity": 0.5,
                  "shadowRadius": 5,
                }
              }
            >
              <RNGestureHandlerButton
                collapsable={false}
                onGestureEvent={[Function]}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                onHandlerStateChange={[Function]}
                rippleColor={0}
              >
                <View
                  accessible={true}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Image
                    source={Object {}}
                    style={
                      Object {
                        "height": 200,
                        "resizeMode": "cover",
                        "width": 300,
                      }
                    }
                  />
                </View>
              </RNGestureHandlerButton>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "padding": 10,
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "paddingBottom": 5,
                        "paddingTop": 3,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "fontSize": 16,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      Mruczek
                       
                    </Text>
                    <Text
                      style={
                        Object {
                          "fontSize": 16,
                        }
                      }
                    >
                      2/3/21
                    </Text>
                  </View>
                  <Text>
                    europejska
                  </Text>
                  <Text>
                    samiec
                    , 
                    5
                     lat, 
                    7
                     kg
                  </Text>
                </View>
                <RNGestureHandlerButton
                  collapsable={false}
                  onGestureEvent={[Function]}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  onHandlerStateChange={[Function]}
                  rippleColor={0}
                >
                  <View
                    accessible={true}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "marginLeft": "35%",
                          "marginTop": 18,
                        }
                      }
                    >
                      <Text />
                    </View>
                  </View>
                </RNGestureHandlerButton>
              </View>
            </View>
          </View>
          <View>
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 15,
                  "elevation": 6,
                  "marginVertical": 20,
                  "overflow": "hidden",
                  "shadowColor": "#fff",
                  "shadowOffset": Object {
                    "height": 10,
                    "width": 10,
                  },
                  "shadowOpacity": 0.5,
                  "shadowRadius": 5,
                }
              }
            >
              <RNGestureHandlerButton
                collapsable={false}
                onGestureEvent={[Function]}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                onHandlerStateChange={[Function]}
                rippleColor={0}
              >
                <View
                  accessible={true}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Image
                    source={Object {}}
                    style={
                      Object {
                        "height": 200,
                        "resizeMode": "cover",
                        "width": 300,
                      }
                    }
                  />
                </View>
              </RNGestureHandlerButton>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "padding": 10,
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "paddingBottom": 5,
                        "paddingTop": 3,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "fontSize": 16,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      Mia
                       
                    </Text>
                    <Text
                      style={
                        Object {
                          "fontSize": 16,
                        }
                      }
                    >
                      1/3/21
                    </Text>
                  </View>
                  <Text>
                    buldog
                  </Text>
                  <Text>
                    samica
                    , 
                    1
                     lat, 
                    5
                     kg
                  </Text>
                </View>
                <RNGestureHandlerButton
                  collapsable={false}
                  onGestureEvent={[Function]}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  onHandlerStateChange={[Function]}
                  rippleColor={0}
                >
                  <View
                    accessible={true}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "marginLeft": "35%",
                          "marginTop": 18,
                        }
                      }
                    >
                      <Text />
                    </View>
                  </View>
                </RNGestureHandlerButton>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
